os/drivers/lcd/st7701.c

```diff
@@ -0,0 +1,362 @@
+	u8 para_list[128];
+} lcm_setting_table_t;
+
+static lcm_setting_table_t ST7701S_init_cmd_g[] = {	/* DCS Write Long */
+	/* ST7701S Reset Sequence */
+	/* LCD_Nreset (1); Delayms (1); */
+	/* LCD_Nreset (0); Delayms (1); */
+	/* LCD_Nreset (1); Delayms (120); */
+	{0x11, 0, {0x00}},
+	{REGFLAG_DELAY, 120, {}},	/* Delayms (120) */
+
+	/* Bank0 Setting */
+	/* Display Control setting */
+	{0xFF, 5, {0x77, 0x01, 0x00, 0x00, 0x10}},
+	{0xC0, 2, {0x63, 0x00}},
+	{0xC1, 2, {0x0C, 0x02}},
+	{0xC2, 2, {0x31, 0x08}},
+	{0xCC, 1, {0x10}},
+
+	/* Gamma Cluster Setting */
+	{0xB0, 16, {0x00, 0x11, 0x19, 0x0C, 0x10, 0x06, 0x07, 0x0A, 0x09, 0x22, 0x04, 0x10, 0x0E, 0x28, 0x30, 0x1C}},
+	{0xB1, 16, {0x00, 0x12, 0x19, 0x0D, 0x10, 0x04, 0x06, 0x07, 0x08, 0x23, 0x04, 0x12, 0x11, 0x28, 0x30, 0x1C}},
+
+	/* End Gamma Setting */
+	/* End Display Control setting */
+	/* End Bank0 Setting */
+
+	/* Bank1 Setting */
+	/* Power Control Registers Initial */
+	{0xFF, 5, {0x77, 0x01, 0x00, 0x00, 0x11}},
+	{0xB0, 1, {0x4D}},
+
+	/* Vcom Setting */
+	{0xB1, 1, {0x3E}},			/* VCOM */
+	/* End End Vcom Setting */
+
+	{0xB2, 1, {0x09}},			/* 07 */
+	{0xB3, 1, {0x80}},
+	{0xB5, 1, {0x4B}},			/* 47 4A */
+	{0xB7, 1, {0x85}},
+	{0xB8, 1, {0x21}},
+	{0xB9, 1, {0x10}},
+	{0xC1, 1, {0x78}},
+	{0xC2, 1, {0x78}},
+	{0xD0, 1, {0x88}},
+	/* End Power Control Registers Initial */
+	{REGFLAG_DELAY, 100, {}},	/* Delayms (100) */
+	/* GIP Setting */
+	{0xE0, 3, {0x00, 0x00, 0x02}},
+	{0xE1, 11, {0x04, 0x00, 0x00, 0x00, 0x05, 0x00, 0x00, 0x00, 0x00, 0x20, 0x20}},
+	{0xE2, 13, {0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}},
+	{0xE3, 4, {0x00, 0x00, 0x33, 0x33}},
+	{0xE4, 2, {0x22, 0x00}},
+	{0xE5, 16, {0x04, 0x34, 0xAA, 0xAA, 0x06, 0x34, 0xAA, 0xAA, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}},
+	{0xE6, 4, {0x00, 0x00, 0x33, 0x00}},
+	{0xE7, 2, {0x22, 0x00}},
+	{0xE8, 16, {0x05, 0x34, 0xAA, 0xAA, 0x07, 0x34, 0xAA, 0xAA, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}},
+	//{0xEA, 16, {0x10, 0x00, 0x10, 0x00, 0x10, 0x00, 0x10, 0x00, 0x10, 0x00, 0x10, 0x00, 0x10, 0x00, 0x10, 0x00}},
+	{0xEB, 7, {0x02, 0x00, 0x40, 0x40, 0x00, 0x00, 0x00}},
+	{0xEC, 2, {0x00, 0x00}},
+	{0xED, 16, {0xFA, 0x45, 0x0B, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xB0, 0x54, 0xAF}},
+	// {0xEF, 12, {0x08, 0x08, 0x08, 0x08, 0x08, 0x08, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04}},
+	{0xEF, 6, {0x08, 0x08, 0x08, 0x44, 0x3F, 0x54}},
+	/* End GIP Setting */
+
+	{0xFF, 5, {0x77, 0x01, 0x00, 0x00, 0x00}},
+	{0x29, 0, {0x00}},
+	{REGFLAG_END_OF_TABLE, 0x00, {}},
+};
+
+static void st7701_send_init_cmd(struct mipi_dsi_device *device, lcm_setting_table_t *table)
+{
+	static u8 send_cmd_idx_s = 0;
+	u32 payload_len;
+	u8 cmd;
+	u8 *cmd_addr;
+
+	struct mipi_dsi_msg msg;
+	while (1) {
+		cmd = table[send_cmd_idx_s].cmd;
+		switch (cmd) {
+		case REGFLAG_DELAY:
+			usleep(table[send_cmd_idx_s].count * 1000);
+			break;
+		case REGFLAG_END_OF_TABLE:
+			printf("LPTX CMD Send Done\n");
```

Comment:
Is this log necessary? Even it is, why not using `lcdvdbg`?

Reply:
I think from information perspective, it would be good idea to print when the last command has been transferred.
It should help debug in future that the bootup sequence is completed.